CMakeLists.txt

```diff
@@ -284,3 +368,16 @@ configure_file(${PROJECT_NAME}.pc.in
                 @ONLY)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
         DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
+
+# Write single include
+# ====================
+
+PREPEND(XTENSOR_SINGLE_INCLUDE "#include <" ${XTENSOR_SINGLE_INCLUDE})
+POSTFIX(XTENSOR_SINGLE_INCLUDE ">" ${XTENSOR_SINGLE_INCLUDE})
+string(REPLACE ";" "\n" XTENSOR_SINGLE_INCLUDE "${XTENSOR_SINGLE_INCLUDE}")
+string(CONCAT XTENSOR_SINGLE_INCLUDE "#ifndef XTENSOR\n" "#define XTENSOR\n\n" "${XTENSOR_SINGLE_INCLUDE}" "\n\n#endif\n")
+
+file(WRITE "${CMAKE_INSTALL_INCLUDEDIR}/xtensor.hpp" "${XTENSOR_SINGLE_INCLUDE}")
+
+install(FILES "${CMAKE_INSTALL_INCLUDEDIR}/xtensor.hpp"
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

Comment:
This can surely be improved, it now creates a temporary file in the repo itself, that is not removed.

Reply:
We could drop the hpp extension so that it becomes

```
#include <xtensor>
```

Reply:
We could drop the hpp extension.so that the include would become

```
#include <xtensor>
```

Reply:
I like it, but CMake refuses to make that file (as it already exists as directory)

Reply:
maybe you could generate the file under `${CMAKE_INSTALL_INCLUDEDIR}/tmp` or `${CMAKE_INSTALL_INCLUDEDIR}/generated` and add this folder to the .gitignore file?

Reply:
That's not a bad idea. My ideas were to:
* Remove it after installation. But I couldn't get it to work, CMake complained that the file could not be installed because it did not exists. 
* A good alternative could be to write this file to the user's build directory. Then we wouldn't have to worry. Here maybe I gave up too quickly, because I cannot remember why I didn't opt for this. 

W.r.t. to @SylvainCorlay 's remark: is there a way to remove the `.hpp` extension?

Reply:
No, it is not possible to have a file and a directory at the same level with the same name. So I think we have to stick with `xtensor.hpp`

Reply:
> A good alternative could be to write this file to the user's build directory.

That's even better than writing the file in a temporary folder in the source tree.

Reply:
OK! Then we should be good to go. I don't think this can be included in the CI, right?